api/app.rb

```diff
@@ -46,34 +46,73 @@ def initialize(*args) super(HTTP_STATUS, *args) end
   end
 end
 
+module BeforeSwitchWorkerUser
+  def self.extended(base)
+    base.instance_exec do
+      helpers do
+        def auth
+          @auth ||= FlightJobScriptAPI.config.auth_decoder.decode(
+            cookies[FlightJobScriptAPI.app.config.sso_cookie_name],
+            env['HTTP_AUTHORIZATION']
+          )
+        end
+
+        def role
+          if auth.valid?
+            :user
+          elsif auth.forbidden?
+            :forbidden
+          else
+            raise Sinja::UnauthorizedError, 'Could not authenticate your authorization credentials'
+          end
+        end
+      end
+
+      # Become the appropriate user if the "role" is user
+      # NOTE: This will trigger the unicorn worker terminator upon finishing the request
+      before do
+        # Only worry about authenticated requests
+        next unless role == :user
```

Comment:
On review this looks wrong, leaving unauthenticated requests as `root` leaves me uneasy.

Probably best to switch to `nobody`.